Client/Extensions/PaymentRequestExtension.swift

```diff
@@ -0,0 +1,142 @@
+                    if error != nil {
+                        log.error(error)
+                    }
+                }
+        }
+    }
+    
+    func userContentController(_ userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
+        guard message.name == PaymentRequestExtension.name(), let body = message.body as? NSDictionary else { return }
+        
+        do {
+            let messageData = try JSONSerialization.data(withJSONObject: body, options: [])
+            let body = try JSONDecoder().decode(PaymentRequest.self, from: messageData)
+            if body.name != "payment-request-show" {
+                return
+            }
+            popup.initPaymentUI()
+            
+            guard body.methodData.contains(where: {$0.supportedMethods == "bat"}) else {
+                sendPaymentRequestError(errorName: PaymentRequestErrors.NotSupportedError.rawValue, errorMessage: Strings.unsupportedInstrumentMessage)
+                return
+            }
+            
+            for item in body.details.displayItems {
+                popup.addDisplayItemLabel(message: "\(item.label): \(item.amount.value) \(item.amount.currency)\n")
+                log.info(item.label)
+            }
+            
+            popup.addTotalLabel(message: "\(body.details.total.label):  \(body.details.total.amount.value) \(body.details.total.amount.currency)\n")
+            
+            popup.addButton(title: Strings.paymentRequestPay) { [weak self] in
+                guard let self = self else {
+                    return .flyDown
+                }
+                guard let rewards = self.rewards, let publisher = self.tab?.publisher, let amount = Double(body.details.total.amount.value) else {
+                    return .flyDown
+                }
+                
+                // To-Do: Replace tipping code with SKU APIs - https://github.com/brave/brave-ios/issues/2383
+                rewards.ledger.tipPublisherDirectly(publisher, amount: amount, currency: "BAT") { _ in
+                    self.response = """
+                        {
+                          "requestId": "a62c29b3-f840-47cd-b895-4573d3190227",
+                          "methodName": "bat",
+                          "details": {
+                            "transaction_id": "bcbbd947-346d-439f-96b4-101bbd966675",
+                            "message": "Payment for Ethiopian Coffee!"
+                          }
+                        }
+                    """
+                    
+                    ensureMainThread {
+                        let trimmed = self.response.components(separatedBy: .newlines).joined()
+                        self.tab?.webView?.evaluateJavaScript("PaymentRequestCallback\(self.token).paymentreq_postCreate('\(trimmed)', '', '')") { _, error in
```

Comment:
same as above

Reply:
The response here will be structured JSON with an `object_id` in place which will be passed from the rewards code. Rewards code is still in progress so the `response` is hard coded for now.